test/unit-tests/vast_test.js

```diff
@@ -584,4 +591,202 @@ describe('ad_manager_vast', function() {
   });
 
   //TODO: Unit test for testing skipoffset with percentage value
+
+  it('Vast 3.0, Error Reporting: Should report too many wrappers error', function(){
+    var embed_code = "embed_code";
+    var vast_ad_mid = {
+      type: "vast",
+      first_shown: 0,
+      frequency: 2,
+      ad_set_code: "ad_set_code",
+      time:10,
+      position_type:"t",
+      url:"1.jpg"
+    };
+    var content = {
+      embed_code: embed_code,
+      ads: [vast_ad_mid]
+    };
+    vastAdManager.initialize(amc);
+    vastAdManager.currentDepth = OO.playerParams.maxVastWrapperDepth;
+    vastAdManager._onVastResponse(vast_ad_mid, wrapperXML);
+    expect(vastAdManager.errorType).to.be("tooManyWrapper");
+    vastAdManager.errorType = '';
+  });
```

Comment:
The unit tests could be made to be a little more robust by checking against the external APIs it will be working with, rather than checking against a property. Do all the errors eventually call OO.pixelPings? If so, you can mock pixelPings to check if it is called properly for these unit tests.